### ISSUE TITLE
Create TokenManager interface for issuing + validating tokens

### ISSUE THREAD

**Issue by nissayeva** [backstage/backstage#7351]
Create a TokenManager interface in `packages/backend-common/src/tokens/types.ts`.

The interface should have two methods, `getServerToken` and `validateToken`.
The `getServerToken` method would for now take no arguments and just return a Promise with a token for backends to use for requests that need authorization. In the future we may want to pass information about the backend itself to this method.
The `validateToken` method should take a token and return a Promise as well, but what should be returned within that Promise is still TBD.

example:
```
interface TokenManager {
  getServerToken: () => Promise<{token: string}>;
  validateToken: (token: string) => Promise<void>;
};
```

**Comment by erikxiv** [comment]
Probably outside of scope for this time, but should it be possible to plugin different auth mechanisms? It is much simpler to build a single mechanism that all adopters use so that is likely a good option now, but there has been a few threads pondering the need for flexibility and how to manage that, such as [this comment](https://github.com/backstage/backstage/issues/5210#issuecomment-812851729). An adopted Backstage instance regularly breaks on updates as many plugin developers do not use auth and forgets to support it. Other adopters want to use their existing auth infrastructure. Future possibilities include traceability, auditability, authorization. In short - there is an option to insert a point of inversion of control for plugin-to-plugin communication (both client/server) that allows adopters to customize transport, authz, logging etc - as these changes would otherwise require updating _all_ plugins from backstage, community and adopters. This would be a bigger scope though, so a question of planning if that kind of breaking change is ripe now.

**Comment by nissayeva** [comment]
the first pass of this is implemented in #7943
